packages/server/migrations/20230914014057_convert-award-ceiling-data-type.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+    return knex.raw(`ALTER TABLE grants ALTER COLUMN award_ceiling TYPE bigint USING award_ceiling::bigint`);
```

Comment:
Note: this does take ~ 10seconds to run locally so I wouldn't suggest to run this migration during heavy usage periods. Will need to be a bit more mindful when we push to production.

Reply:
FWIW, I think your idea of using a multi-step migration would mitigate the downtime concerns – you could adjust a bit to something like this:
1. Add new bigint column named `award_ceiling_temp` using values cast from the existing `award_ceiling` column
2. Rename columns (single transaction):
    - `award_ceiling` -> `award_ceiling_old`
    - `award_ceiling_temp` -> `award_ceiling`
3. Drop column `award_ceiling_old` (note: you might be able to do this in step 2 instead of renaming to `award_ceiling_old`)

This "add+populate+rename+drop" operation is a pretty typical pattern for "altering" existing columns on a high-traffic DB.

Reply:
Also, if we're moving `award_ceiling` to bigint, what do you think about doing the same for `award_floor` – I would imagine it would have the same sorting issues. 

A bigint might not be as necessary for storing `award_floor` values, but might be worth doing anyways for consistency and/or "just in case".

Reply:
> This "add+populate+rename+drop" operation is a pretty typical pattern for "altering" existing columns on a high-traffic DB.

Agreed on this in theory. Should we do that for our current application given the usage patterns?

Reply:
Resolved in a separate PR.